### PR TITLE
extend: Add instance for Function

### PR DESCRIPTION
**Pull request by gabejohnson**
~~I could add the `Comonad` instance as well if anyone's interested~~  😦 

Actually I can't as `extract` can't take a `TypeRep`.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -1261,6 +1261,7 @@ test('extend', function() {
   eq(Z.extend(joinWith(''), ['x', 'y']), ['xy', 'y']);
   eq(Z.extend(joinWith(''), ['x', 'y', 'z']), ['xyz', 'yz', 'z']);
   eq(Z.extend(function(id) { return Z.reduce(add, 1, id); }, Identity(42)), Identity(43));
+  eq(Z.extend(function(f) { return f([1, 2]) + '2'; }, function(xs) { return xs.length; })([3, 4]), '42');
```
Comment (davidchambers) [inline test/index.js]: Let's replace the second function expression with `length` (defined near the top of the file).

File: index.js [hunk 2]
```diff
@@ -1110,6 +1118,7 @@
         'fantasy-land/promap':      Function$prototype$promap,
         'fantasy-land/ap':          Function$prototype$ap,
         'fantasy-land/chain':       Function$prototype$chain,
+        'fantasy-land/extend':      Function$prototype$extend,
```
Comment (davidchambers) [inline index.js]: Please also make this addition:

```diff
   //. `fantasy-land/extend` implementations are provided for the following
-  //. built-in types: Array.
+  //. built-in types: Array and Function.
```

Comment (davidchambers) [reply index.js]: A doctest would be useful too. :)

File: test/index.js (hunk 1 above)
Comment (davidchambers) [reply test/index.js]: Actually, the expression above relies on type coercion. Let's avoid this. I'd also like the expression to demonstrate the order of arguments to `concat`. I suggest:

```javascript
> Z.extend(function(f) { return f([3, 4]); }, Z.reverse)([1, 2])
[4, 3, 2, 1]
```

Further improvements much appreciated. :)

File: index.js (hunk 2 above)
Comment (gabejohnson) [reply index.js]: Done.

File: test/index.js (hunk 1 above)
Comment (gabejohnson) [reply test/index.js]: I can't think of any at the moment.

**Pull request merged**